gradio/components/login_button.py

```diff
@@ -0,0 +1,86 @@
+"""Predefined button to sign in with Hugging Face in a Gradio Space."""
+from __future__ import annotations
+
+from typing import Any, Literal
+
+from gradio_client.documentation import document, set_documentation_group
+
+from gradio.components import Button
+from gradio.routes import Request
+
+set_documentation_group("component")
+
+
+@document()
+class LoginButton(Button):
+    """
+    Button that redirects the user to Sign with Hugging Face using OAuth.
+    """
+
+    is_template = True
+
+    def __init__(
+        self,
+        *,
+        value: str = "Sign in with Hugging Face",
+        variant: Literal["primary", "secondary", "stop"] = "secondary",
+        size: Literal["sm", "lg"] | None = None,
+        visible: bool = True,
+        interactive: bool = True,
+        elem_id: str | None = None,
+        elem_classes: list[str] | str | None = None,
+        scale: int | None = 0,
+        min_width: int | None = None,
+        _activate: bool = True,
```

Comment:
Why do we have this parameter?

Reply:
I have genuinely no idea :smile: 
I took it from [`gr.DuplicateButton`](https://github.com/gradio-app/gradio/blob/f5539c7618e31451420bd3228754774da14dc65f/gradio/components/duplicate_button.py#L37) and thought it was an internal Gradio magic. Should I remove it?

Reply:
Yes its not really needed even in the case of `gr.DuplicateButton`, and is not needed here. I would just call `self.activate()` directly from the constructor.